docs/api/1.x/authentication.rst

```diff
@@ -356,3 +361,24 @@ The ``parseToken()`` function scans the location hash to read the Identity Provi
     }
 
 Check out the :github:`full demo source code <leplatrem/kinto-oidc-demo>`.
+
+
+Example of 0Auth dance redirections
+-----------------------------------
+
+Let's assume the JavaScript app is accessible on http://localhost:3000 and the Kinto server running on http://localhost:8888.
+
+When the user clicks the login button, the browser will follow a sequence of redirections similar to this one:
+
+#. User clicks on the ``auth0`` login button
+#. JavaScript redirects to `<http://localhost:8888/v1/openid/auth0/login?scope=openid email&callback=http://localhost:3000/#provider=auth0&tokens=>`_
+#. Kinto generates and stores a *state* string
+#. Kinto redirects to Auth0 that will show the login form `<https://minimal-demo-iam.auth0.com/authorize?client_id=BXqGVgl2meRsdVK0dEZPTk516JUhje2M&response_type=code&scope=openid+email&redirect_uri=http://localhost:8888/v1/openid/auth0/token?&state=3a309f5baba>`_
+#. User enters credentials and authenticates
+#. Auth0 redirects to Kinto with the *state* and a *code* `<http://localhost:8888/v1/openid/auth0/token?code=lWpsu9VoHLJEVyy1&state=3a309f5baba>`_
+#. Kinto checks that the *state* matches
+#. Kinto trades the *code* against the ID and Access tokens
+#. Kinto redirects back to the Single Page App appending the JSON encoded ID and Access tokens to the callback URL provided at step 2 `<http://localhost:3000/#provider=auth0&tokens={"access_token":"tY6um989...jfcer","id_token":"eyJ0eXAiOiJ...KV1QiL.ojkhgwRVH...UG8JGRENNF.Es8...DK10","expires_in":86400,"token_type":"Bearer"}>`_
+#. JavaScript code parses the location hash and reads the ID and Access tokens
```

Comment:
I'm curious now, how would it reach log files? If the token is sent as hash location the browser does not put it in the http request AFAIK

Reply:
I checked my Nginx server and it doesn't write down anything in the log files after the `#` character. Because, I guess, the browser doesn't even bother sending that as part of the URL to server. Nothing to worry about then.

Reply:
I knew it thanks to ZeroBin :) https://news.ycombinator.com/item?id=3832269